autolink does not work expected

OS: Ubuntu 16.04
Zulip: 1.7.1

autolink does not work that I expected like following example:

```
日本語http://example.com/
```

This URL with preceding multibyte characters.

expected result:

```
日本語<a href="http://example.com/">http://example.com/</a>
```

I found large regular expression as below:

https://github.com/zulip/zulip/blob/83f563aeff2358ba159b2349ab39382450e376a9/zerver/lib/bugdown/__init__.py#L1401-L1422

And I've created small patch for this issue:

```diff
$ git diff master
diff --git a/zerver/lib/bugdown/__init__.py b/zerver/lib/bugdown/__init__.py
index b1434c724..eb905fae8 100644
--- a/zerver/lib/bugdown/__init__.py
+++ b/zerver/lib/bugdown/__init__.py
@@ -1421,7 +1421,7 @@ class Bugdown(markdown.Extension):
         nested_paren_chunk = nested_paren_chunk % (inner_paren_contents,)
         tlds = '|'.join(list_of_tlds())
         link_regex = r"""
-            (?<![^\s'"\(,:<])    # Start after whitespace or specified chars
+            (?<![^\s\w'"\(,:;<])    # Start after whitespace or specified chars
                                  # (Double-negative lookbehind to allow start-of-string)
             (?P<url>             # Main group
                 (?:(?:           # Domain part
diff --git a/zerver/tests/test_bugdown.py b/zerver/tests/test_bugdown.py
index e8fab4c0c..b05c7bc33 100644
--- a/zerver/tests/test_bugdown.py
+++ b/zerver/tests/test_bugdown.py
@@ -1104,6 +1104,17 @@ class BugdownTest(ZulipTestCase):
             '<p><a href="http://example.com/#settings/" target="_blank" title="http://example.com/#settings/">http://example.com/#settings/</a></p>'
         )
 
+    def test_normal_link_start_with_non_ascii_character(self) -> None:
+        realm = get_realm("zulip")
+        sender_user_profile = self.example_user('othello')
+        message = Message(sender=sender_user_profile, sending_client=get_client("test"))
+        msg = "あhttp://example.com/#settings/"
+
+        self.assertEqual(
+            bugdown.convert(msg, message_realm=realm, message=message),
+            '<p>あ<a href="http://example.com/#settings/" target="_blank" title="http://example.com/#settings/">http://example.com/#settings/</a></p>'
+        )
+
     def test_relative_link(self) -> None:
         realm = get_realm("zulip")
         sender_user_profile = self.example_user('othello')
```

But this change will break a test like following:

```
FAIL: test_bugdown_fixtures (zerver.tests.test_bugdown.BugdownTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/python/3.4.6/lib/python3.4/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/opt/python/3.4.6/lib/python3.4/unittest/case.py", line 580, in run
    testMethod()
  File "/home/travis/build/okkez/zulip/zerver/tests/test_bugdown.py", line 233, in test_bugdown_fixtures
    self.assertEqual(converted, test['expected_output'])
  File "/opt/python/3.4.6/lib/python3.4/unittest/case.py", line 800, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/python/3.4.6/lib/python3.4/unittest/case.py", line 1173, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/opt/python/3.4.6/lib/python3.4/unittest/case.py", line 645, in fail
    raise self.failureException(msg)
AssertionError: '<p>&[17 chars]p://x<a href="http://x.com/xss.js&lt;b" target[68 chars]</p>' != '<p>&[17 chars]p://xx.com/xss.js&lt;b&gt;</p>'
- <p>&lt;script src=http://x<a href="http://x.com/xss.js&lt;b" target="_blank" title="http://x.com/xss.js&lt;b">x.com/xss.js&lt;b</a>&gt;</p>
+ <p>&lt;script src=http://xx.com/xss.js&lt;b&gt;</p>
```

What should I do?

Thanks.

--- a/zerver/lib/markdown/__init__.py
+++ b/zerver/lib/markdown/__init__.py
@@ -259,8 +259,9 @@ def get_web_link_regex() -> Pattern[str]:
     nested_paren_chunk %= (inner_paren_contents,)
 
     file_links = r"| (?:file://(/[^/ ]*)+/?)" if settings.ENABLE_FILE_LINKS else r""
+
     REGEX = rf"""
-        (?<![^\s'"\(,:<])    # Start after whitespace or specified chars
+        (?<![^\s'"\(,:<\u0080-\U0010FFFF])    # Start after whitespace, specified chars, or multibyte chars
                              # (Double-negative lookbehind to allow start-of-string)
         (?P<url>             # Main group
             (?:(?:           # Domain part

--- a/zerver/tests/fixtures/markdown_test_cases.json
+++ b/zerver/tests/fixtures/markdown_test_cases.json
@@ -1492,6 +1492,21 @@
       "asanadesktop://task/123456789",
       "<p>%s</p>",
       "asanadesktop://task/123456789"
+    ],
+    [
+      "日本語https://google.com",
+      "<p>日本語<a href=\"https://google.com\">https://google.com</a></p>",
+      "https://google.com"
+    ],
+    [
+      "你好https://google.com",
+      "<p>你好<a href=\"https://google.com\">https://google.com</a></p>",
+      "https://google.com"
+    ],
+    [
+      "Приветhttps://google.com",
+      "<p>Привет<a href=\"https://google.com\">https://google.com</a></p>",
+      "https://google.com"
     ]
   ]
 }

